src/ParseNode.js

```diff
@@ -290,13 +290,15 @@ type ParseNodeTypes = {
  * typing. Throws if the node's type does not match.
  */
 export function assertNodeType<NODETYPE: NodeType>(
-    node: ParseNode<*>,
+    // The union allows either ParseNode<*> or the union of two specific nodes.
+    node: ?ParseNode<*> | ParseNode<*>,
```

Comment:
Shouldn't `ParseNode<A> | ParseNode<B>` be subtype of `ParseNode<*>`?

Reply:
That's what I originally thought, but then I remembered that `*` doesn't mean "anything". It means "something specific that I'd like the inference to figure out". The constraint in `ParseNode`'s type parameter is that it's one of the strings in `NodeType`. So it shouldn't be able to correspond to two possible values.

Reply:
More info here in the Existential type is here:
https://flow.org/en/docs/types/utilities/#toc-existential-type
Note that sometimes the type inference can't figure out what the Existential type should be since it does *global* type inference and in my experience kinda just constraints the possibilities that there type can have. That's why we were able to get away a lot more with just using `*` for the input if `htmlBuilder` and `mathmlBuilder` before we ported everything to flow. So I generally recommend always specifying explicit types whenever possible -- I've caught more type errors due to the increased strictness that way.

Reply:
More info here in the Existential type is here:
https://flow.org/en/docs/types/utilities/#toc-existential-type
Note that sometimes the type inference can't figure out what the Existential type should be since it does *global* type inference and in my experience kinda just constraints the possibilities that there type can have. That's why we were able to get away a lot more with just using `*` for the input if `htmlBuilder` and `mathmlBuilder` before we ported everything to flow. So I generally recommend always specifying explicit types whenever possible -- I've caught more type errors due to the increased strictness that way.

Reply:
Sorry for the late comment.  Isn't
```
node: ?ParseNode<*> | ParseNode<*>
```
equivalent to
```
node: ?ParseNode<*>
```
?

Reply:
@kevinbarabash As @marcianx said above, from what I understand, `*` doesn't mean "anything", rather it is inferred to one (at the compile-time). So `?ParseNode<*>` cannot accept `ParseNode<A> | ParseNode<B>` as `*` can be either A or B but it cannot be determined at the compile-time.

Reply:
```
node: ?ParseNode<*>
``` 
should be equivalent to 
```
node: ParseNode<*> | null | undefined
```
I would expect:
```
node: ?ParseNode<*> | ParseNode<*>
```
to be equivalent to:
```
node: ParseNode<*> | null | undefined | ParseNode<*>
```
but since there's two `ParseNode<*>` in the same union we should be able to drop one result in:
```
node: ParseNode<*> | null | undefined
```
which is equivalent to
```
node: ?ParseNode<*>
```

Reply:
To elaborate on ylemkimon@'s comment, if `*` meant "anything", then yes, `ParseNode<*>` is equivalent to `ParseNode<*> | ParseNode<*>` because each `ParseNode<*>` there is the *same*. But in flow, each `*` is an *existential* type, which means that *each* `*` refers to something in particular (in our case, it's constrainted to be a value from `NodeType`) that the flow type system has to infer. The two `*`s may be inferred to be *different* things as they are independent and not connected in any way, and thus the union is strictly more generic.

More concretely, let's say you are the flow type inference system and you see a parameter of type `ParseNode<*>`. If a caller passes in something of type `ParseNode<"accent"> | ParseNode<"supsub">`, what specific value is `*` inferred to be?
However, if the parameter has type `ParseNode<*> | ParseNode<*>`, then the first `*` can be inferred to be "accent", and the second one can be inferred to be "supsub".